Fix all bugs on login and logout

vbertoletti opened the pull request:
This PR fixes bugs on login screen, profile screen and realm/user provider, as they are all connected to login and logout. Profile page renders data from db without errors.

vbertoletti committed:
work in progress for the signout call to realm method

vbertoletti committed:
small change

siddharthparmar7 commented on client/js/screens/LogIn/LogInContainer.js:
```diff
@@ -22,11 +22,13 @@ class LoginContainer extends Component {
-            {({ signin, token }) => {
-              if (token) {
+            {({ storeSessionToken, token }) => {
+              if (token.length > 0) {
```
if you make your Context state null initially you will not have to check for your token's length. You can just do `if(token)`

siddharthparmar7 commented on client/js/screens/LogIn/LogInContainer.js:
```diff
@@ -22,11 +22,13 @@ class LoginContainer extends Component {
-                return <LogIn login={login} signin={signin} />;
+                return (
+                  <LogIn login={login} storeSessionToken={storeSessionToken} />
+                );
```
add prop type validation.

siddharthparmar7 replied:
this is not resolved yet.

siddharthparmar7 commented on client/js/screens/Profile/ProfileContainer.js:
```diff
@@ -21,7 +35,39 @@ class ProfileContainer extends Component {
+          }
+        }}
+      </UserContext.Consumer>
+    );
```
please add prop type validations.

siddharthparmar7 replied:
Please resolve this.

siddharthparmar7 commented on client/js/screens/Profile/Profile.js:
```diff
@@ -51,11 +57,12 @@ const Profile = ({ navigation }) => {
+          logout(id);
         }}
       >
         <Text style={styles.buttonText}>LOGOUT</Text>
```
add prop type validation.

siddharthparmar7 reviewed (changes requested):
I can not check your app because it throws an error Unable to resolve module `../screens/Login` from root stack navigation. Please fix it. Also, you should be calling your dirs, files and components SignIn and SignUp, not LogIn and SignUp. Please make the changes. Thank you.

vbertoletti commented:
Fixed all requirements, thanks.

siddharthparmar7 commented on client/js/context/UserContext/UserProvider.js:
```diff
@@ -6,34 +6,44 @@ class UserProvider extends Component {
-      token: "Test Token",
-      token: "",
+      token: null,
       id: ""
```
make this null too.

siddharthparmar7 commented on client/js/context/UserContext/UserProvider.js:
```diff
@@ -6,34 +6,44 @@ class UserProvider extends Component {
+    try {
+      await Realm.write(() => {
+        Realm.delete(Realm.objectForPrimaryKey("User", id));
+        this.setState({ id: "", token: null });
```
id should be null too.

siddharthparmar7 replied:
ID is not null here.

siddharthparmar7 commented on client/js/screens/Profile/Profile.js:
```diff
@@ -64,3 +72,10 @@ const Profile = ({ navigation }) => {
+  navigation: PropTypes.object,
+  data: PropTypes.object,
+  logout: PropTypes.func,
+  id: PropTypes.string
```
the props that are not required, you should have default props for them.

siddharthparmar7 commented on client/js/screens/SignIn/SignIn.js:
```diff
@@ -10,10 +10,11 @@ import {
+const SignIn = ({ login, storeSessionToken }) => {
   return (
     <Fragment>
       <StatusBar barStyle="light-content" />
```
make all your View components Fragment that is just by themselves.

siddharthparmar7 commented on client/js/screens/SignIn/SignIn.js:
```diff
@@ -111,4 +112,9 @@ const LogIn = ({ login, storeSessionToken }) => {
+
+SignIn.propTypes = {
+  login: PropTypes.func,
+  storeSessionToken: PropTypes.func
```
add default props if these props are not required.

siddharthparmar7 reviewed (changes requested):
Please resolved all the issues. Thank you.

vbertoletti commented:
Amended proptypes as required, id initial state is now null, also added fragment to one section that was wrapped in unnecessary view tag, fixed LogInContainer to be SignInContainer.

siddharthparmar7 commented on client/js/screens/Profile/ProfileContainer.js:
```diff
@@ -21,8 +35,46 @@ class ProfileContainer extends Component {
+ProfileContainer.defaultProps = {
+  removeUserIdToken: null,
+  id: null,
+  data: null
```
these are not props. They are function arguments. 

the only prop here is navigation.

siddharthparmar7 commented on client/js/screens/SignIn/SignInContainer.js:
```diff
@@ -36,4 +38,10 @@ class LoginContainer extends Component {
+SignInContainer.defaultProps = {
+  login: null,
+  storeSessionToken: null,
+  token: null
```
these are not props. these are function arguments.

You should have all of your prop validations with .isRequired and if they are not required only then have default props.

siddharthparmar7 reviewed (changes requested):
Please make the requested changes and don't validate your function arguments. Validate your props.

vbertoletti commented:
Thanks for taking the time to clarify, I have pushed more changes. Please let me know if something needs improvements.

vbertoletti committed:
intial work on getting the challenges progression in the profile page

vbertoletti committed:
solve merge conflicts

vbertoletti committed:
solve merge conflicts

vbertoletti committed:
solve merge conflicts

vbertoletti committed:
fix all merge conflicts and re-add some code lost in the process, also re-name files once again to match component

siddharthparmar7 reviewed (approved):
Thank you!
